lib/hdk_time_indexing/src/sorting.rs

```diff
@@ -0,0 +1,62 @@
+use std::cmp::Ordering;
+use chrono::{DateTime, Utc, NaiveDateTime};
+use hdk::prelude::*;
+use crate::{
+    TimeIndexResult,
+    reading::link_prefix_for_index,
+    index_tree::IndexSegment,
+};
+
+/**
+ * Higher-order function to generate a comparator callback for use in
+ * sorting sets of `EntryHash`es that have been indexed into the
+ * `index_name` time-ordered index.
+ *
+ * It is presumed that the entries are validly linked in the index. If
+ * not, they will be placed at the end of the list in undefined order.
+ *
+ * Usage:
+ *  let mut my_entry_hashes: Vec<EntryHash> = some_list_of_hashes;
+ *  my_entry_hashes.sort_by(sort_entries_by_time_index("my_time_index_name"));
+ */
+pub fn sort_entries_by_time_index<'a, I>(index_name: &'a I) -> Box<dyn for<'r> Fn(&EntryHash, &EntryHash) -> Ordering + 'a>
```

Comment:
I'm feeling on the fence about this function. It occurs to me we could be doing the `get_links` call multiple times per entry_hash in the list of addresses. Could we, in the code that calls this function, instead call `get_time_for_entry_hash` a guaranteed once per address in the list, and then use that list of date structs paired with their entry hash to sort by?

Reply:
We could, but I don't think it's necessary for now.

a) Rust does seem to do quite intelligent optimisations for core-level sorting APIs and the compiler might be optimising some of this for us; and
b) It does seem like (based on conversations on Discord with Paul D'Aoust) Holochain subconscious should optimise this for us and be fetching links efficiently from its local cache upon subsequent reads for the same data.